catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/security/ShareableMetacardPolicyExtension.java

```diff
@@ -125,14 +134,26 @@ private KeyValueCollectionPermission isPermitted(
 
     // filter out all implied permissions
     Function<Set<String>, KeyValueCollectionPermission> filterPermissions =
-        (implied) -> {
-          List<KeyValuePermission> values =
-              permissions
+        (impliedKeys) -> {
+          KeyValueCollectionPermission implied =
+              new KeyValueCollectionPermission(
+                  match.getAction(),
+                  impliedKeys
+                      .stream()
+                      .map(key -> new KeyValuePermission(key, grouped.get(key)))
+                      .collect(Collectors.toList()));
+
+          List<Permission> remaining =
+              allPerms
+                  .<Permission>getKeyValuePermissionList()
                   .stream()
-                  .filter((permission) -> !implied.contains(permission.getKey()))
+                  .filter(p -> !implied.implies(p))
                   .collect(Collectors.toList());
 
-          return new KeyValueCollectionPermission(match.getAction(), values);
+          KeyValueCollectionPermission collection =
+              new KeyValueCollectionPermission(match.getAction());
```

Comment:
Does the `KeyValueCollectionPermission` constructor not take a collection as a second argument?

Reply:
It does, but the signature matches the impl not the interface: 
```
public KeyValueCollectionPermission(String action, Collection<KeyValuePermission> permissions)
```